src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/ConstantExpectedAnalyzer.cs

```diff
@@ -0,0 +1,566 @@
+                case SpecialType.System_Int16:
+                    return UnmanagedHelper<long>.Validate(parameterSymbol, attributeData, short.MinValue, short.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_Int32:
+                    return UnmanagedHelper<long>.Validate(parameterSymbol, attributeData, int.MinValue, int.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_Int64:
+                    return UnmanagedHelper<long>.Validate(parameterSymbol, attributeData, long.MinValue, long.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_IntPtr:
+                    return UnmanagedHelper<long>.Validate(parameterSymbol, attributeData, int.MinValue, int.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_Single:
+                    return UnmanagedHelper<float>.Validate(parameterSymbol, attributeData, float.MinValue, float.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_Double:
+                    return UnmanagedHelper<double>.Validate(parameterSymbol, attributeData, double.MinValue, double.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_Boolean:
+                    return UnmanagedHelper<bool>.Validate(parameterSymbol, attributeData, false, true, Helper, out diagnostics);
+                case SpecialType.System_String:
+                    return Validate(parameterSymbol, attributeData, Helper, out diagnostics);
+                case SpecialType.None when parameterSymbol.Type.TypeKind == TypeKind.TypeParameter:
+                    return Validate(parameterSymbol, attributeData, Helper, out diagnostics);
+                default:
+                    diagnostics = Helper.ParameterIsInvalid(parameterSymbol.Type.ToDisplayString(), attributeData.ApplicationSyntaxReference.GetSyntax());
+                    return false;
+            }
+        }
+
+        private static bool TryGetConstantExpectedAttributeData(IParameterSymbol parameter, [NotNullWhen(true)] out AttributeData? attributeData)
+        {
+            AttributeData? constantExpectedAttributeData = parameter.GetAttributes()
+                .FirstOrDefault(attrData => IsConstantExpectedAttribute(attrData.AttributeClass));
+            attributeData = constantExpectedAttributeData;
+            return constantExpectedAttributeData is not null;
+        }
+
+        private static bool HasConstantExpectedAttributeData(IParameterSymbol parameter)
+        {
+            return parameter.GetAttributes()
+                .Any(attrData => IsConstantExpectedAttribute(attrData.AttributeClass));
+        }
+
+        private static bool IsConstantExpectedAttribute(INamedTypeSymbol namedType)
+        {
+            return namedType.Name.Equals(ConstantExpectedAttribute, StringComparison.Ordinal) &&
```

Comment:
We should do symbol check for the attribute, the attribute should be loaded on the `CompilationStart` and stop analyzing further if the attribute doesn't exit for the compilation. Probably you could reference the attribute from within the repo or continue using the mock attribute for testing

Reply:
The attribute symbol is now used for comparison